d2bs/kolbot/libs/common/Misc.js

```diff
@@ -7,22 +7,2305 @@
 
 const Skill = {
 	usePvpRange: false,
-	haveTK: false,
+	skills: {
+		6: {
+			hardpoints: false,
+			checked: false,
+			have: function () {
+				if (this.hardpoints) return true;
```

Comment:
these have functions are largely repetitive, can you replace them with single impl where the skill is an arg?
in the cases where you have a config check, you can wrap it:

```
have: function() {
   if (condition) return false;
   return have_common();
}
```

Reply:
Done, that cleaned it up a lot